src/aggregate.jl

```diff
@@ -0,0 +1,320 @@
+
+const DimOrDimTuple = Union{Dimension,Tuple{Vararg{<:Dimension}}}
+const IntOrIntTuple = Union{Int,Tuple{Vararg{<:Int}}}
+
+struct Ag end
+struct DisAg end
+
+struct Every{V,S}
+    val::V
+    start::S
+end
+Every(val; start=nothing) = Every(val, start)
+
+
+const SCALE = """
+- `scale`: the aggregation factor, which can be an `Integer`, a tuple of `Integer`
+    for each dimension, or a `Dimension`
+"""
+
+const METHOD = """
+- `method`: a function such as `mean` or `sum` that can combine the
+    value of multiple cells to generate the aggregated cell, or a `Symbol` 
+    `:start`, `:center`, or `:end` or `Locus` like `Start()` `Center()` or `End`
```

Comment:
Why not use `first` and `last` instead of `:start` and `:end`? Even `:center` could be replaced with a new function (even if the underlying implemention is more efficient).

Reply:
That makes sense, we can just define `center`. 

I didnt make them functions because they wouldnt actually be used as functions of the windiw area like `sum`, and because with reversed order lookups they will actually give you the oppodite value to calling `first` on the window.